community/pmc-tool/collector/main.py

```diff
@@ -1,61 +1,58 @@
-#
-#   http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing,
-# software distributed under the License is distributed on an
-# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-# KIND, either express or implied.  See the License for the
-# specific language governing permissions and limitations
-# under the License.
-
-"""This file contains logic for interfacing with GitHub and fetching PR and issue data."""
-
-
-from db import create_db_connection, create_pull_requests_table, insert_prs_to_db, fetch_pull_requests
-from gh import query_prs, load_token, get_api_endpoint
-
-# Set up authentication with access token
-headers = load_token('token.txt')
-
-# Set up API endpoint
-API_ENDPOINT = get_api_endpoint()
+from db import create_db_connection, create_pull_requests_table, insert_pull_requests_to_db, \
+    fetch_pull_requests, create_issues_table, fetch_issues, insert_issues_to_db
+from gh import query_prs, query_issues
 
 # Connect to the SQLite database
 conn = create_db_connection('pull_requests.db')
 
 # Create a table to store pull requests if it doesn't exist
 create_pull_requests_table(conn)
+create_issues_table(conn)
 
 # Retrieve list of pull requests
-pull_requests = query_prs(API_ENDPOINT, headers)
+pull_requests = query_prs()
+issues = query_issues()
 
 # Extract list of pull request authors
-authors_count = {}
+pr_authors_count = {}
 
 for pr in pull_requests:
     author = pr['user']['login']
-    if author not in authors_count:
-        authors_count[author] = 0
-    authors_count[author] += 1
+    if author not in pr_authors_count:
+        pr_authors_count[author] = 0
+    pr_authors_count[author] += 1
 
-# Insert pull request data into the database
-insert_prs_to_db(conn, prs=[{'author': author, 'count': count} for author, count in authors_count.items()])
+# Extract list of issue authors
+issue_authors_count = {}
+
+for issue in issues:
+    author = issue['user']['login']
+    if author not in issue_authors_count:
+        issue_authors_count[author] = 0
+    issue_authors_count[author] += 1
+
+# Insert pull request and issue data into the database
+pr_data_to_insert = [{'author': author, 'count': count} for author, count in pr_authors_count.items()]
+insert_pull_requests_to_db(conn, prs=pr_data_to_insert)
+
+issue_data_to_insert = [{'author': author, 'count': count} for author, count in issue_authors_count.items()]
+insert_issues_to_db(conn, issues=issue_data_to_insert)
 
 # Fetch and print pull request data from the database
-results = fetch_pull_requests(conn)
-for row in results:
+pr_results = fetch_pull_requests(conn)
+print("Pull Requests:")
+for row in pr_results:
     author, count = row
     print(f'Author: {author}, PRs: {count}')
 
+# Fetch and print issue data from the database
+issue_results = fetch_issues(conn)
+print("Issues:")
+for row in issue_results:
+    author, count = row
+    print(f'Author: {author}, Issues: {count}')
```

Comment:
I like this idea, but do not print the data - instead, let's add a 'verification function'. Something like this:

```py
def assert_equal(db_data, api_data):
  db_set = set(db_data)
  api_set = set(api_data.items())
  if db_set != api_set:
    raise ValueError('The data from the database does not match the data from the API. '
        + 'Database has %s extra. ' % db_set - api_set
        + 'Api has %s extra.' % api_set - db_set)
```

```suggestion
pr_results = fetch_pull_requests(conn)
print("Validating Pull Requests...")
assert_equal(pr_results, pr_authors_count)

# Fetch and print issue data from the database
issue_results = fetch_issues(conn)
print("Issues:")
assert_equal(issue_results, issue_authors_count)
```